10666 Add button to configure Related resource relationship type

whatisgalen opened the pull request:
<!--- Provide a general summary of the Pull Request in the Title above -->
### Types of changes
<!--- Put an `x` in the boxes that apply  -->
-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change
<!--- Include a brief description of this Pull Request and reasoning behind it. -->


### Issues Solved
<!--- If this Pull Request solves any issues, please list them here  -->
#10666

### Checklist
<!--- Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.  -->
-   [x] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

#### Ticket Background
*   Sponsored by: @scholiumtech
*   Found by: @whatisgalen 
*   Tested by: @jacobtylerwalls 
*   Designed by: @whatisgalen 

### Further comments

<!--- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->

jacobtylerwalls commented on arches/app/templates/views/components/datatypes/resource-instance.htm:
```diff
@@ -35,13 +35,19 @@
                             onEnterkeyClick, onSpaceClick," 
-                        style="cursor: pointer;">
+                            style="cursor: pointer;">
+                            <i class="fa fa-gear"></i>
```
This has meaning, so let's add an `aria-label`. Maybe there is a suitable string already in javascript.htm like `edit`?

whatisgalen replied:
requested changes in latest

jacobtylerwalls commented on arches/app/templates/views/resource/related-resources/related-resources-manager.htm:
```diff
@@ -138,7 +138,12 @@ <h3 class="">
-                            <div class="rr-table-column" style="flex-grow: 1;" data-bind="click: self.toggleSelectedResourceRelationship, clickBubble: false">
+                            <div class='rr-table-column icon-column'>
+                                <button data-bind="click: self.toggleSelectedResourceRelationship, clickBubble: false">
+                                    <i class="fa fa-gear"></i>
```
same

jacobtylerwalls commented on arches/app/templates/views/components/datatypes/resource-instance.htm:
```diff
@@ -35,13 +35,19 @@
+                            data-bind="
                             click: function() { if($parent.isEditable !== false){self.toggleSelectedResource($data)}; },
                             clickBubble: false,
                             onEnterkeyClick, onSpaceClick," 
```
With this change, the space action no longer works; it just flickers:


https://github.com/archesproject/arches/assets/38668450/da170fad-a622-4867-983e-fd648fa3e9d7

jacobtylerwalls replied:
Still seeing this in the latest, did you have a look?

I'm showing this is fixed with
```diff
diff --git a/arches/app/templates/views/components/datatypes/resource-instance.htm b/arches/app/templates/views/components/datatypes/resource-instance.htm
index 60d6eb9f7..eceb1ff07 100644
--- a/arches/app/templates/views/components/datatypes/resource-instance.htm
+++ b/arches/app/templates/views/components/datatypes/resource-instance.htm
@@ -41,8 +41,7 @@
                             data-bind="
                             click: function() { if($parent.isEditable !== false){self.toggleSelectedResource($data)}; },
                             attr: {'aria-label': $root.translations.configureRelatedResourceRelationship, 'disabled': $parent.isEditable === false,},
-                            clickBubble: false,
-                            onEnterkeyClick, onSpaceClick," 
+                            clickBubble: false,"
                             style="cursor: pointer;">
                             <i class="fa fa-gear"></i>
                         </button>
```

whatisgalen replied:
fix in latest

jacobtylerwalls commented on arches/app/templates/views/components/datatypes/resource-instance.htm:
```diff
@@ -35,13 +35,19 @@
-                        style="cursor: pointer;">
+                            style="cursor: pointer;">
+                            <i class="fa fa-gear"></i>
+                        </button>
```
When the node is not editable, we should style the icon as disabled, just like delete:

<img width="590" alt="disabled-style" src="https://github.com/archesproject/arches/assets/38668450/100458cd-6391-4f98-830d-1eadebf0004a">

whatisgalen replied:
fix in latest

chiatt commented:
@whatisgalen, Could you rename this PR to something more descriptive, e.g. `Button for resource instance relationship type config`?

jacobtylerwalls commented on arches/app/templates/javascript.htm:
```diff
@@ -189,6 +189,7 @@
         select-a-nodegroup='{% trans "Select a nodegroup" as selectANodegroup %} "{{ selectANodegroup|escapejs }}"'
         no-relationships-added='{% trans "No Relationships Added" as noRelationshipsAdded %} "{{ noRelationshipsAdded|escapejs }}"'
         relate-resource='{% trans "Relate Resource" as relateResource %} "{{ relateResource|escapejs }}"'
+        configure-related-resource-relationship='{% trans "Configure Related Resource Relationship" as configureRelateResourceRelationship %} "{{ configureRelateResourceRelationship|escapejs }}"'
```
Just a heads up: the left side is `related` and the right side is `relate`: would be great to have both the same

whatisgalen replied:
Whoops that is definitely a typo

whatisgalen commented:
I think there are some other bugs related to this UI. When I tried testing as of the latest merge from `dev/7.6.x`, the first issue is that this check in the resource-instance widget template  `<!-- ko if: self.relationship() -->` was returning false (undefined) for no apparent reason. Both resource models had ontologies set up and are relatable to each other. (This persisted when I switched from the feature branch to `dev/7.6.x`) When I removed that check from the htm file, the valueid for the relationship type appeared instead of the preflabel.

<img width="1483" alt="Screenshot 2024-03-18 at 4 33 41 PM" src="https://github.com/archesproject/arches/assets/8366985/ed1bad94-78c1-4299-b6fc-e3ef973851cf">

jacobtylerwalls reviewed (changes requested):
Thanks for updates, just two little nits and we can get this in 👍

jacobtylerwalls commented:
Couple more things just occurred to me.

- Would be nice to remove the cursor pointer and tabindex from the resource label if it's no longer interactive
- Did you consider making these changes in resource-instance-creator.htm also?

whatisgalen commented:
> Couple more things just occurred to me.
> 
> * Would be nice to remove the cursor pointer and tabindex from the resource label if it's no longer interactive ✅ 
> * Did you consider making these changes in resource-instance-creator.htm also?

Is there a separate UX for creating resource-instance datatype data in `related-instance-creator.js`? I'm seeing that it just uses the existing widgets.


https://github.com/archesproject/arches/assets/8366985/585ee71e-a861-410c-9fff-19c20dfc1a77

jacobtylerwalls reviewed (approved):
+1 tested. Thanks for the updates!
